In [1]:
import pandas as pd
import numpy as np

## read and merge dicom download files

Load the CSV files from the DICOM download, image collection creation process on LONI IDA.

In [2]:
ad2_m_mri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI2/010_ADNI2_M_MRI_8_02_2021.csv')
ad2_m_fmri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI2/012_ADNI2_M_fMRI_8_04_2021.csv')
ad2_f_mri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI2/009_ADNI2_F_MRI_8_02_2021.csv')
ad2_f_fmri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI2/011_ADNI2_F_fMRI_8_04_2021.csv')

ad3_m_mri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI3/ADNI_3_MRI_M_8_06_2021.csv')
ad3_m_fmri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI3/ADNI_3_fMRI_M_8_06_2021.csv')
ad3_f_mri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI3/ADNI_3_MRI_F_8_06_2021.csv')
ad3_f_fmri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNI3/ADNI_3_fMRI_F_8_06_2021.csv')

adgo_m_mri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNIGO/002_GO_M_MRI_8_02_2021.csv')
adgo_m_fmri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNIGO/006_GO_M_fMRI_8_02_2021.csv')
adgo_f_mri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNIGO/001_GO_F_MRI_8_02_2021.csv')
adgo_f_fmri = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/ADNIGO/005_GO_F_fMRI_8_02_2021.csv')

ad3_u = pd.read_csv('/N/project/statadni/20231212_ADR012021_UtahBackup/ImagingData/dicomImages/Saige_20250410/Wolfers_ADNI3_SinceJune21_10_06_2025.csv')

In [3]:
# merge all files into one big dataframe.
ad2_m = pd.merge(ad2_m_mri, ad2_m_fmri, how='outer')
ad2_f = pd.merge(ad2_f_mri, ad2_f_fmri, how='outer')
ad2 = pd.merge(ad2_m, ad2_f, how='outer')
ad3_m = pd.merge(ad3_m_mri, ad3_m_fmri, how='outer')
ad3_f = pd.merge(ad3_f_mri, ad3_f_fmri, how='outer')
ad3 = pd.merge(ad3_m, ad3_f, how='outer')
adgo_m = pd.merge(adgo_m_mri, adgo_m_fmri, how='outer')
adgo_f = pd.merge(adgo_f_mri, adgo_f_fmri, how='outer')
adgo = pd.merge(adgo_m, adgo_f, how='outer')
ad23 = pd.merge(ad2, ad3, how='outer')

In [4]:
# fix merge error by changing visit code to object dtype.
ad23['Visit'] = ad23['Visit'].astype('object')

In [5]:
# fix merge error by changing visit code to object dtype.
adgo['Visit'] = adgo['Visit'].astype('object')

In [6]:
# merge all files into one big dataframe.
ad233 = pd.merge(ad23, ad3_u, how='outer')
all_data = pd.merge(ad233, adgo, how='outer')

In [7]:
# Check the number of DICOM directories based on the LONI IDA Image collection csv files.
len(all_data)

83726

In [8]:
# Check the number of unique subjects. 
len(all_data['Subject'].value_counts())

2018

^ Based on the CSV files from the LONI image collection downloads, there are **83726 DICOM directories** from **2018 unique subjects**.

In [9]:
# There are 501 different directory names for the DICOM directories.
all_data['Description'].value_counts()

Description
3 Plane Localizer                   11795
localizer                            6779
Field Mapping                        5348
MPRAGE                               3454
Axial T2-FLAIR                       3043
                                    ...  
Axial rsfMRI (Eyes Open) 10 min         1
Axial FLAIR REPEAT                      1
Axial T2-TSE with Fat Sat repeat        1
FLAIR AX                                1
Axial 2D PASL_RPT_AP216                 1
Name: count, Length: 501, dtype: int64

In [10]:
# Build a dictionary of all possible fMRI DICOM directory names.
fmri_names = ['Axial rsfMRI (Eyes Open)',
'Resting State fMRI',
'Axial MB rsfMRI (Eyes Open)',
'Axial fcMRI (EYES OPEN)',
'Axial rsfMRI (EYES OPEN)',
'Axial fcMRI (Eyes Open)',
'Extended Resting State fMRI',
'Axial rsfMRI (Eyes Open) -phase P to A',
'Axial fcMRI 0 angle (EYES OPEN)',
'Axial fcMRI',
'Axial MB rsfMRI (Eyes Open)   straight no angle',
'Axial RESTING fcMRI (EYES OPEN)',
'Axial MB rsfMRI AP',
'Axial rsfMRI (Eyes Open) 10 min :-PJ',
'Extended AXIAL rsfMRI EYES OPEN',
'Axial rsfMRI (Eyes Open) Phase Direction P>A',
'AXIAL RS fMRI (EYES OPEN)',
'Axial_rsFMRI_Eyes_Open',
'Extended Resting State fMRI CLEAR',
'Axial MB rsfMRI (Eyes Open) REPEAT',
'Axial fcMRI (EYES OPEN)_REPEAT',
'Axial rsfMRI P>>A (Eyes Open)',
'Axial MB rsfMRI',
'Axial MB rsfMRI (Eyes Open) (MSV21)',
'epi_2s_resting_state',
'Axial rsfMRI (Eyes Open) 10 min']

In [11]:
fmri_all_data = all_data[all_data['Description'].isin(fmri_names)]

In [12]:
# Check the number of fMRI DICOM directories.
fmri_all_data.shape

(3244, 12)

In [13]:
# Check the number of unique subjects who have an fMRI DICOM directory.
len(fmri_all_data['Subject'].value_counts())

1291

There are **3244 fMRI DICOM directories** from **1291 unique subjects** (based on the LONI IDA Image Collection csv files).

In [41]:
# save the merge DICOM directory file.
all_data.to_csv('/N/project/statadni/20250922_Saige/adni_db/csv/ALL_IMG.csv', index=False)

In [14]:
# Load the file I created from just listing the unzipped DICOM directories.
# I did this in an attempt to verify the download. 
# This list should match the all_data list if the download was successful.
dcm_df = pd.read_csv('/N/project/statadni/20250922_Saige/QC/dcm_directories.csv')

In [15]:
# Check the number of DICOM directories.
dcm_df.shape

(29551, 7)

In [16]:
# Check the number of unique subjects. 
len(dcm_df['Subject_ID'].value_counts())

2018

^ Based on the file I created by listing the unzipped DICOM directories, there are **29551 DICOM directories** from **2018 unique subjects**.

The number of unique subjects (2018) matches, but there seems to be a lot more DICOM directories in the LONI IDA csv files than in the unzipped DICOM directories. 

This discrepancy may be due to Vincent dowloading more modalities (i.e., DTI) than we are currently using. So maybe he didn't unzip these files. In the image collection that I (Saige) built, I queried on ADNI3 subjects (uploaded to LONI between June 2021 and October 2025) who have both MRI AND fMRI. I think Vincent just made an image collection of all data so he downloaded some DICOM directories/files that we aren't actually using.

In [17]:
# There are only 483 different names of the DICOM directories, compared to 501 names in the step above, so something is a bit off. 
dcm_df['scan_name'].value_counts()

scan_name
3_Plane_Localizer                   1793
Sagittal_3D_FLAIR                   1428
Perfusion_Weighted                  1185
Field_Mapping                       1102
Axial_T2_Star                       1072
                                    ... 
AXIAL_T2_STAR_230_FOV                  1
MP-RAGE_-_REPEAT                       1
Axial_-_Advanced_fMRI_64_Channel       1
Field_Mapping_2.5mm                    1
Axial_rsfMRI_P__A__Eyes_Open_          1
Name: count, Length: 483, dtype: int64

In [19]:
# This cell was used to list out the DICOM directory names.
# Then I had to manually go through and chose the names that are for fMRI scans, which I input to the dictionary in the cell below.
#pd.set_option('display.max_rows', None)

In [18]:
# Build a dictionary of all the fMRI DICOM directory names
fmri_dcm = ['Axial_rsfMRI__Eyes_Open_','Resting_State_fMRI','Axial_MB_rsfMRI__Eyes_Open_', 'Axial_fcMRI__EYES_OPEN_','Axial_rsfMRI__EYES_OPEN_',
'Extended_Resting_State_fMRI','Axial_fcMRI__Eyes_Open_', 'Axial_rsfMRI__Eyes_Open__-phase_P_to_A', 'Axial_RESTING_fcMRI__EYES_OPEN_',
'Axial_rsfMRI__Eyes_Open__10_min__-PJ','Axial_MB_rsfMRI__Eyes_Open____straight_no_angle','Axial_rsfMRI__Eyes_Open__Phase_Direction_P_A',
'Extended_AXIAL_rsfMRI_EYES_OPEN','AXIAL_RS_fMRI__EYES_OPEN_', 'Axial_rsFMRI_Eyes_Open','Axial_MB_rsfMRI__Eyes_Open__REPEAT',
'Axial_rsfMRI__Eyes_Open__10_min','epi_2s_resting_state','Axial_fcMRI__EYES_OPEN__REPEAT','Axial_MB_rsfMRI__Eyes_Open___MSV21_',
'Axial_MB_rsfMRI','Axial_-_Advanced_fMRI_64_Channel','Axial_rsfMRI_P__A__Eyes_Open_']

In [19]:
# Build a new dataframe that only contains the fMRI DICOM directories.
fmri_dcm_df = dcm_df[dcm_df['scan_name'].isin(fmri_dcm)]

In [20]:
# Check the number of fMRI DICOM directories.
fmri_dcm_df.shape

(1901, 7)

In [21]:
# Check the number of unique subjects with fMRI DICOM directory.
len(fmri_dcm_df['Subject_ID'].value_counts())

1284

^ Based on the file I created by listing the unzipped DICOM directories, there are **1901 fMRI DICOM directories** from **1284 unique subjects**.

In [22]:
# This is the count of how many DICOM fMRI directories come from each study.
# ADNI3_MRI_fMRI_F = Vincent download, ADNI3 Females
# ADNI3_MRI_fMRI_M = Vincent download, ADNI3 Males
# ADNI3_update = Saige download, ADNI3 M&F (updated data since Vincent's download)
# ADNI2_proc_MRI_fMRI_F = Vincent download, ADNI2 Females
# ADNI2_proc_MRI_fMRI_M = Vincent download, ADNI2 Males
# ADNIGO_MRI_fMRI_F = Vincent download, ADNIGO Females
# ADNIGO_MRI_fMRI_M = Vincent download, ADNIGO Males
fmri_dcm_df['study_ID'].value_counts()

study_ID
ADNI3_MRI_fMRI_F         574
ADNI3_MRI_fMRI_M         508
ADNI3_update             460
ADNI2_proc_MRI_fMRI_F    174
ADNI2_proc_MRI_fMRI_M    158
ADNIGO_MRI_fMRI_F         20
ADNIGO_MRI_fMRI_M          7
Name: count, dtype: int64

In [ ]:
# add line here to make adni_subs.txt file to pass to Clinica script. 